Example Knative PowerShell Email Function (#380)
vmware-samples/vcenter-event-broker-appliance: examples/knative/powershell/kn-ps-email/README.md

# kn-ps-email
Example Knative PowerShell function that uses [Send-MailKitMessage](https://www.powershellgallery.com/packages/Send-MailKitMessage) to send email notification.

![](screenshots/screenshot-1.png)

# Step 1 - Build

Create the container image and optionally push to an external registry such as Docker.

```
docker build -t <docker-username>/kn-ps-email:1.0 .
docker push <docker-username>/kn-ps-email:1.0
```

# Step 2 - Test

Verify the container image works by executing it locally.

Change into the `test` directory
```bash
cd test
```

Update the following variable names  within the `docker-test-env-variable` file

* SMTP_SERVER - Address of Email Server
* SMTP_PORT - Port of Email Server
* SMTP_USERNAME - Username to send email (use empty string for no username)

> * SMTP_PASSWORD - Password to send emaail (use empty string for no username)

Reviewer: ```suggestion
* SMTP_PASSWORD - Password to send email (use empty string for no password)
```

```

Testing Function ...
See docker container console for output

# Output from docker container console
Server start listening on 'http://*:8080/'
DEBUG: K8s Secrets:
....
DEBUG: CloudEventData
 {
  "Vm": {
    "Vm": {
      "Type": "VirtualMachine",
      "Value": "vm-10838"
    },
    "Name": "FooBar"
  },
  "Key": 2040013,
  "Ds": null,
  "Dvs": null,
  "CreatedTime": "2021-04-23T20:21:02.277Z",
  "Net": null,
  "ChainId": 2040012,
  "ComputeResource": {
    "ComputeResource": {
      "Type": "ClusterComputeResource",
      "Value": "domain-c8"
    },
    "Name": "Supermicro-Cluster"
  },
  "UserName": "VSPHERE.LOCAL\\william",
  "ChangeTag": "",
  "Datacenter": {
    "Datacenter": {
      "Type": "Datacenter",
      "Value": "datacenter-3"
    },
    "Name": "Primp-Datacenter"
  },
  "Host": {
    "Host": {
      "Type": "HostSystem",
      "Value": "host-11"
    },
    "Name": "192.168.30.5"
  },
  "Template": false,
  "FullFormattedMessage": "Removed FooBar on 192.168.30.5 from Primp-Datacenter"
}

Sending Secure Email ...
```

# Step 3 - Deploy

> **Note:** The following steps assume a working Knative environment using the

> `default` Rabbit broker. The Knative `service` and `trigger` will be installed in the

Reviewer: ```suggestion
`default` Rabbit `broker`. The Knative `service` and `trigger` will be installed in the
```

> `vmware-functions` Kubernetes namespace, assuming that the broker is also available there.

Reviewer: ```suggestion
`vmware-functions` Kubernetes namespace, assuming that the `broker` is also available there.
```

> Update the `email_secret.json` file with your email configurations and then create the kubernetes secret which can then accessed from within the function by using the environment variable named called `EMAIL_SECRET`.

Reviewer: ```suggestion
Update the `email_secret.json` file with your email configurations and then create the kubernetes secret which can then be accessed from within the function by using the environment variable named called `EMAIL_SECRET`.
```
